src/server/login.js

```diff
@@ -53,14 +51,27 @@ module.exports = function (client, server, options) {
 
       try {
         const publicKey = crypto.createPublicKey({ key: packet.signature.publicKey, format: 'der', type: 'spki' })
-        const publicPEM = mcPubKeyToPem(packet.signature.publicKey)
-        const signable = packet.signature.timestamp + publicPEM // (expires at + publicKey)
-
-        if (!crypto.verify('RSA-SHA1', Buffer.from(signable, 'utf8'), mojangPubKey, packet.signature.signature)) {
-          raise('multiplayer.disconnect.invalid_public_key_signature')
-          return
+        const publicDER = publicKey.export({ type: 'spki', format: 'der' })
+        const publicPEM = mcPubKeyToPem(publicDER)
+        if (mcData.supportFeature('profileKeySignatureV2')) {
+          if (!packet.playerUUID) {
+            client.end('Invalid packet format', JSON.stringify({ translate: 'disconnect.genericReason', with: [{ text: 'Invalid packet format' }] }))
```

Comment:
You can use the raise function

Reply:
Raise does not allow for translation with arguments.